### PR TITLE
Hw 6

### PR DESCRIPTION
**Pull request by setday**

### CODE REVIEW

**Comment by setday** [comment]
hw-6 пока не готова. Проверять не надо.

File: notebooks/06-homework.ipynb [hunk 1]
```diff
@@ -0,0 +1,1961 @@
...
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "x_train, x_test, y_train, y_test = train_test_split(tfid_result, df['rating'], train_size=0.8, random_state=113)\n",
+    "\n",
+    "def plot_bar(y, loc='left', relative=True):\n",
```
Comment (svkov) [inline notebooks/06-homework.ipynb]: А чем тебе стандартный hist не угодил? Если хотелось именно в процентах посчитать, есть параметр `density`

File: notebooks/06-homework.ipynb [hunk 2]
```diff
@@ -0,0 +1,1961 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "a7f14e34",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tfid = TfidfVectorizer()\n",
+    "tfid_result = tfid.fit_transform(df['text']).toarray()"
```
Comment (svkov) [inline notebooks/06-homework.ipynb]: тут ты переобучаешься - сразу обучаешься и на трейне и на тесте

File: notebooks/06-homework.ipynb [hunk 3]
```diff
@@ -0,0 +1,1961 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "fd94f6c4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def show_accuracy(y_test, y_pred):\n",
```
Comment (svkov) [inline notebooks/06-homework.ipynb]: почему функция называется show_accuracy, а внутри используешь f1?

misleading название, сложно читать такое)

File: notebooks/06-homework.ipynb [hunk 4]
```diff
@@ -0,0 +1,1961 @@
...
...pjgAAAAAs0pAhQzJ48OBG+1q1arXIxw0cODB/+9vf8qc//WlplfaxBF8AAAAALFKrVq0WK+j6sGOPPTYjRozIo48+mlVXXbVhf/fu3TN79uxMmjSpUdfX+PHj071794Y1Tz75ZKPn++BTHz9YsyhudQQAAABgiaqvr8+xxx6bu+66Kw8//HB69erV6Pjmm2+eFi1a5KGHHmrY9/LLL2fMmDGpra1NktTW1ub555/P22+/3bDmgQceSIcOHdKnT5/FqkPHFwAAAEAZSilVuoRl3sCBA3PzzTfn7rvvTvv27RtmcnXs2DGtW7dOx44dc8QRR2Tw4MHp3LlzOnTokOOOOy61tbXp27dvkmS33XZLnz59cvDBB2f48OEZN25cTj311AwcOHCxO89K9fX19UvtVVbIjDmFe0kALCWz586vdAmwzJk0Y06lS4Bl0oTJi/fJZbA82XLNjpUuoSL+8s/JlS6hIrZaa/Hf71Jp4eHg9ddfn8MOOyxJMmvWrJx00km55ZZbUldXl9133z2XXXZZo9sYX3/99QwYMCB//OMf07Zt2xx66KE555xz0rz54vVyCb4AWK4JvmBBgi9YOMEXLEjwtXwpJ/haVpjxBQAAAEAhCb4AAAAAKCTD7QEAAADK8DHjq1gG6fgCAAAAoJAEXwAAAAAUkuALAAAAgEISfAEAAABQSIbbAwAAAJTBbPvqoeMLAAAAgEISfAEAAABQSIIvAAAAAApJ8AUAAABAIRluDwAAAFAO0+2rho4vAAAAAApJ8AUAAABAIQm+AAAAACgkwRcAAAAAhWS4PQAAAEAZSqbbVw0dXwAAAAAUkuALAAAAgEISfAEAAABQSGZ8AQAAAJShZMRX1dDxBQAAAEAhCb4AAAAAKCTBFwAAAACFJPgCAAAAoJAMtwcAAAAog9n21UPHFwAA8H/t3XuUlQW9N/Dv5jYid1RABBQlSU+KKUZ4Q3RS02OS9nYsL1h2McFQ1IJMTTMhS0PFwMrEUhRTQV81jehFtKUIGB41xMvBFOOmCArCgDPz/uE6c84ElWPCnnn8fNaatXiu+7dnrc2e/d2/5/cAQCEJvgAAAAAoJMEXAAAAAIUk+AIAAACgkAy3BwAAAGgI0+2bDB1fAAAAABSS4AsAAACAQhJ8AQAAAFBIgi8AAAAACslwewAAAIAGKJlu32To+AIAAACgkARfAAA
```
Comment (svkov) [inline notebooks/06-homework.ipynb]: а что ты тут предсказываешь регрессией?

деревья усредняют значения в листьях, и на выходе ты можешь получить невалидное значение (например, 2.2)
так можно делать, но нужно обоснование, почему ты захотел так сделать, и, наверное, не через f1 такое измерять надо

в классе мы как-то решали задачу классификации, а метрику брали из регрессии. такой трюк тут можно проворачивать без вопросов, но модель все-таки лучше брать классификации

Comment (svkov) [inline notebooks/06-homework.ipynb:1551]: пока что ты просто подобрал несколько отзывов, не хватает обоснование, какие слова стоит использовать, а какие нет. посмотри на importance модели, там разгадка

**Review by svkov — CHANGES_REQUESTED** [review]
На medium сделано больше, чем нужно, и сделано неправильно. Но раз тебя интересует тема регрессии, то пересчитай модели классификации с метрикой регрессии и посмотри, насколько результаты похожи на f1.

Пока что только easy засчитываю)

File: notebooks/06-homework.ipynb (hunk 3 above)
Comment (setday) [reply notebooks/06-homework.ipynb]: Ну, хотелось абстрагировать функцию от метрики. `accuracy` тут используется не в значении метрики, в значении `точность модели`

File: notebooks/06-homework.ipynb [hunk 5]
```diff
@@ -0,0 +1,1765 @@
...
...BqgZLZ9k6HjCwAAAIBCEnwBAAAAUEiCLwAAAAAKyYwvAAAAgAYw4qvp0PEFAAAAQCEJvgAAAAAoJMEXAAAAAIUk+AIAAACgkAy3BwAAAGgI0+2bDB1fAAAAABSS4AsAAACAQhJ8AQAAAFBIgi8AAAAACslwewAAAIAGKJlu32To+AIAAACgkARfAAAAABSS4AsAAACAQhJ8AQAAAFBIhtsDAAAANEDJbPsmQ8cXAAAAAIUk+AIAAACgkARfAAAAABSS4AsAAACAQjLcHgAAAKABzLZvOnR8AQAAAFBIgi8AAAAACknwBQAAAEAhCb4AAAAAKCTD7QEAAAAawnT7JkPHFwAAAACb1Jw5czJkyJB07949pVIp06dPr3e8trY2F198cbbffvu0bt06gwcPzvPPP19vzapVq3Lcccelffv26dixY04++eSsXbu2QXUIvgAAAADYpNatW5e+fftm/Pjx73j86quvzvXXX5+JEyfm8ccfT5s2bXL44Ydn/fr1dWuOO+64PPPMM5k5c2ZmzJiROXPm5JRTTmlQHaXa2traf+lKGqG1VYW7JNgkWjTXjwt/b13Vm+UuARqdhcvWlbsEaJQGfvbCcpcAjc4bvx5X7hLK4sU/rv/Hiwpol21a/VOvK5VKmTZtWoYOHZrkrW6v7t2755xzzsm5556bJFmzZk26du2aKVOm5Nhjj838+fOz++6754knnki/fv2SJPfff38++clPZsmSJenevfv7em8dXwAAAAANUPo3/W9TWbRoUZYtW5bBgwfX7evQoUP69++fRx99NEny6KOPpmPHjnWhV5IMHjw4zZo1y+OPP/6+38twewAAAAD+oaqqqlRVVdXbV1FRkYqKigadZ9myZUmSrl271tvftWvXumPLli1Lly5d6h1v0aJFOnfuXLfm/dDxBQAAAMA/VFlZmQ4dOtT7U1lZWe6y3pOOLwAAAAD+odGjR2fUqFH19jW02ytJunXrliRZvnx5tt9++7r9y5cvz1577VW3ZsWKFfVe9+abb2bVqlV1r38/dHwBAAAA8A9VVFSkffv29f78M8FXz549061bt8yaNatu32uvvZbHH388AwYMSJIMGDAgq1evzty5c+vWPPjgg6mpqUn//v3f93vp+AIAAABogNKmm/NeWGvXrs3ChQvrthctWpR58+alc+fO6dGjR0aOHJnLL788H
```
Comment (svkov) [inline notebooks/06-homework.ipynb:1645]: > это было реально очевидно ввиду структуры модели

если бы в исходных данных было много примеров с иронией, модель могла бы негативные слова записать в позитивные. так мы лишний раз убедились, как именно работает модель.

плюс здесь довольно очевидная предметная область. в 8 домашке, например, если поставить себе цель обмануть модель, то без заглядывания в importance это будет сделать проблематично.

Comment (svkov) [inline notebooks/06-homework.ipynb]: мы еще никаких нейронок не обучали =)

File: notebooks/06-homework.ipynb [hunk 6]
```diff
@@ -0,0 +1,1765 @@
...
+    "    print(f1_score(y_test, y_pred, average='weighted'))\n",
+    "\n",
+    "    cm = confusion_matrix(y_test, y_pred)\n",
+    "    plt.figure(figsize=(16, 16))\n",
+    "    sns.heatmap(cm, annot=True, fmt='d', cmap='Blues')\n",
+    "    plt.xlabel('Predicted')\n",
+    "    plt.ylabel('Actual')\n",
+    "    plt.show()\n",
+    "\n",
+    "def show_regression_model_perfection_metrics(y_test, y_pred):\n",
```
Comment (svkov) [inline notebooks/06-homework.ipynb]: вот эту функцию можно использовать, чтобы измерять качество вообще всех моделей в этой задаче. а модели регрессии использовать тут мало смысла, заранее известно, что они будут хуже работать, чем модели классификации.

**Review by svkov — APPROVED** [review]
Зачтено
